src/ledger/LedgerTxnImpl.h

```diff
@@ -267,13 +269,35 @@ class LedgerTxn::Impl
     std::shared_ptr<LedgerEntry const>
     getNewestVersion(LedgerKey const& key) const;
 
+    // dropFromEntryCacheIfPresent has the strong exception safety guarantee.
+    // If it throws an exception, the state of the LedgerTxnRoot entry cache
+    // will be unchanged.
+    void dropFromEntryCacheIfPresent(LedgerKey const& key);
+
     // load has the basic exception safety guarantee. If it throws an exception,
     // then
     // - the prepared statement cache may be, but is not guaranteed to be,
     //   modified
     // - the entry cache may be, but is not guaranteed to be, cleared.
     LedgerTxnEntry load(LedgerTxn& self, LedgerKey const& key);
 
+    // createOrUpdateWithoutLoading has the basic exception safety guarantee.
+    // If it throws an exception, then
+    // - the entry cache in the parent LedgerTxnRoot may, but is not guaranteed
+    //   to, have the given entry removed.
+    // - the current LedgerTxn::Impl may, but is not guaranteed to, have
```

Comment:
I'm not sure what you mean here. Do you mean that I'm referring to the entry cache being cleared? that's residue from previous versions of this that did entry-cache clearing. I'll remove that. if you mean that it only presents the basic exception guarantee, I'm confused: create also only has the basic exception guarantee.

With the revised semantics of only populating the entry map (not changing the active map) suggests though that this can be made strong. Erase cannot, I don't think (assuming we want it to continue to affect both the active and entry maps).

Reply:
discussed in-chat, resolved.